### DISCUSSION TITLE
Storing User Agent

### DISCUSSION THREAD

**Discussion by chrgeor** [umami-software/umami#2711] [General]
I run umami for a few years now and looking back to my stats I suspected that some traffic is from bots. I thought that the detect mechanism could not work for new user agents. So my plan was to somehow reconcile the data. To my surprise there is no user-agent stored for the sessions! Is this by design?

**Comment by mikecao** [comment]
Yes, we do not store user agents due to privacy.

**Comment by chrgeor** [comment]
Ok I agree, but are there resources that define user-agent as private data or is an arbitrary architectural decision?
